### PR TITLE
add complement test for getting joined_members after leaving.

### PR DESCRIPTION
**Pull request by andrewdoh**
Signed-off-by: Andrew Doh <andrewddo@gmail.com>

### CODE REVIEW

File: tests/csapi/apidoc_room_state_test.go [hunk 1]
```diff
@@ -320,5 +321,18 @@ func TestRoomState(t *testing.T) {
...
+		t.Run("GET /rooms/:room_id/joined_members test to ensure user cannot request joined_members after leaving room", func(t *testing.T) {
+			t.Parallel()
+			roomID := authedClient.CreateRoom(t, map[string]interface{}{})
+			authedClient.LeaveRoom(t, roomID)
+			res := authedClient.DoFunc(t, "GET", []string{"_matrix", "client", "r0", "rooms", roomID, "joined_members"})
+			must.MatchResponse(t, res, match.HTTPResponse{
+				StatusCode: http.StatusForbidden,
+				JSON: []match.JSON{
+					match.JSONKeyEqual("errcode", "M_FORBIDDEN"),
+					match.JSONKeyEqual("error", "Getting joined members after leaving is forbidden."),
```
Comment (DMRobertson) [inline tests/csapi/apidoc_room_state_test.go:333]: The error message here isn't part of the spec---other homeservers could choose a different message (e.g. localised for the requester's language). I think we should remove this line.

```suggestion
```

File: tests/csapi/apidoc_room_state_test.go [hunk 2]
```diff
@@ -320,5 +321,18 @@ func TestRoomState(t *testing.T) {
 				},
 			})
 		})
+		t.Run("GET /rooms/:room_id/joined_members test to ensure user cannot request joined_members after leaving room", func(t *testing.T) {
```
Comment (DMRobertson) [inline tests/csapi/apidoc_room_state_test.go:324]:
```suggestion
		t.Run("GET /rooms/:room_id/joined_members is forbidden after leaving room", func(t *testing.T) {
```

**Review by DMRobertson — CHANGES_REQUESTED** [review]

**Review by DMRobertson — APPROVED** [review]
LGTM!

**Comment by DMRobertson** [comment]
Will probably need matrix-org/synapse#13374 before merging.

**Comment by babolivier** [comment]
It looks like this PR is causing issue with the CI:

```
tests/csapi/apidoc_room_state_test.go:16:2: http redeclared as imported package name
	/home/runner/work/complement/complement/tests/csapi/apidoc_room_state_test.go:4:2: previous declaration
```

This seems to be because https://github.com/matrix-org/complement/pull/415 added an import of `net/http` to this file. @andrewdoh can you fix this please? Happy to merge this PR once it's done (and CI is happy)

**Comment by andrewdoh** [comment]
> It looks like this PR is having issue with the CI:
> 
> ```
> tests/csapi/apidoc_room_state_test.go:16:2: http redeclared as imported package name
> 	/home/runner/work/complement/complement/tests/csapi/apidoc_room_state_test.go:4:2: previous declaration
> ```
> 
> This seems to be because #415 added an import of `net/http` to this file. @andrewdoh can you fix this please? Happy to merge this PR once it's done (and CI is happy)

yes, its fixed. you're right, looks like we both imported http per #415.

**Comment by andrewdoh** [comment]
hmmm, I suppose this means that its not working for golang implementation of the matrix protocol? I could add it tomorrow.

**Comment by kegsay** [comment]
You are correct, please `runtime.SkipIf(runtime.Dendrite)`

**Comment by andrewdoh** [comment]
I've updated the PR with the suggested, thanks @kegsay.

**Review by DMRobertson — APPROVED** [review]

**Pull request merged**